### PR TITLE
Modal enhancement

### PR DESCRIPTION
**Pull request by wagner-schwertz-poatek**
added: close button to modal component
added: smoother animations to modal component
added: ability trigger onClose on key press

Added navigation features as well as smoother animations (including animation when closing modal

https://user-images.githubusercontent.com/84329762/146301145-39dff18d-7d76-4554-9f85-478e76161c5c.mp4

)

### CODE REVIEW

**Review by jotape-poatek — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [3.0.1] 2021-12-15
```
Comment (gateixeira) [inline CHANGELOG.md:10]: I think this should be a minor instead of a patch, provided it's an enhancement rather than a fix.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@farmerconnect/farmerconnect-ui",
-  "version": "3.0.0",
+  "version": "3.0.1",
```
Comment (gateixeira) [inline package.json:3]: 3.1.0

**Review by gateixeira — CHANGES_REQUESTED** [review]
Please check the versioning

**Review by tiagoalmeidadarosa — DISMISSED** [review]

**Comment by wagner-schwertz-poatek** [comment]
the changes in this PR are not suited for the current direction of the Modal component

**Pull request closed without merging**